pkg/store/device/store.go

```diff
@@ -116,3 +127,45 @@ func (s *topoStore) Watch(ch chan<- *devicepb.Device) error {
 func getTopoConn(opts ...grpc.DialOption) (*grpc.ClientConn, error) {
 	return grpc.Dial(topoAddress, opts...)
 }
+
+// DeviceConnected signals the corresponding topology service that the device connected.
+func (s *topoStore) DeviceConnected(id devicepb.ID) (*devicepb.Device, error) {
+	log.Infof("Device %s connected", id)
+	return s.updateDevice(id, devicepb.ConnectivityState_REACHABLE, devicepb.ChannelState_CONNECTED,
+		devicepb.ServiceState_AVAILABLE)
+}
+
+// DeviceDisconnected signal the corresponding topology service that the device disconnected.
+func (s *topoStore) DeviceDisconnected(id devicepb.ID, err error) (*devicepb.Device, error) {
+	log.Infof("Device %s disconnected or had error in connection %s", id, err)
+	//TODO check different possible availabilities based on error
+	return s.updateDevice(id, devicepb.ConnectivityState_UNREACHABLE, devicepb.ChannelState_DISCONNECTED,
+		devicepb.ServiceState_UNAVAILABLE)
+}
+
+func (s *topoStore) updateDevice(id devicepb.ID, connectivity devicepb.ConnectivityState, channel devicepb.ChannelState,
+	service devicepb.ServiceState) (*devicepb.Device, error) {
+	getResponse, err := s.requestClient.Get(context.Background(), &devicepb.GetRequest{
+		ID: id,
+	})
+	if err != nil {
+		return nil, err
+	}
+	topoDevice := getResponse.Device
+	protocolState := new(devicepb.ProtocolState)
+	protocolState.Protocol = devicepb.Protocol_GNMI
+	protocolState.ConnectivityState = connectivity
+	protocolState.ChannelState = channel
+	protocolState.ServiceState = service
+	topoDevice.Protocols = append(topoDevice.Protocols, protocolState)
```

Comment:
will check for existence. 

regarding the identity, in our TST discussion we agreed not to pass that to TOPO because topo is oblivious to these kind of things, it cares about a service, not the master for that service.